cartography/intel/aws/sagemaker/domains.py

```diff
@@ -0,0 +1,148 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import boto3
+import neo4j
+
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from cartography.models.aws.sagemaker.domain import AWSSageMakerDomainSchema
+from cartography.util import aws_handle_regions
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_domains(
+    boto3_session: boto3.session.Session,
+    region: str,
+) -> List[Dict[str, Any]]:
+    """
+    Get all SageMaker Domains in the given region.
+    """
+    client = boto3_session.client("sagemaker", region_name=region)
+    paginator = client.get_paginator("list_domains")
+    domains: List[Dict[str, Any]] = []
+
+    # Get all domain IDs
+    domain_ids: List[str] = []
+    for page in paginator.paginate():
+        for domain in page.get("Domains", []):
+            domain_ids.append(domain["DomainId"])
+
+    # Get detailed information for each domain
+    for domain_id in domain_ids:
+        try:
```

Comment:
I don't think we need to have a `try+except` here because aws_handle_regions takes care of it.